### PR TITLE
Added initial version of tables, table_info and table_data methods

### PR DESCRIPTION
**Pull request by djberg96**
This PR allows us to get at the storage account tables and read them. These tables are what store various bits of metrics data in Azure. You can get a list of the tables with the .tables method, or actually read and filter them with the .table_data method.

I also included some basic specs.

### CODE REVIEW

File: lib/azure/armrest/model/storage_account.rb [hunk 1]
```diff
@@ -24,6 +28,52 @@ def initialize(json)
         @storage_api_version = '2015-02-21'
       end
 
+      # Returns a list of tables for the given storage account +key+. Note
+      # that full metadata is returned.
+      #
+      def tables(key = nil)
```
Comment (bzwei) [inline lib/azure/armrest/model/storage_account.rb]: with our naming convention, this method better to be `list_tables`

File: lib/azure/armrest/model/storage_account.rb [hunk 2]
```diff
@@ -24,6 +28,52 @@ def initialize(json)
...
+        key ||= properties.key1
+        response = table_response(key, nil, "Tables")
+        JSON.parse(response.body)['value'].map{ |t| Table.new(t) }
+      end
+
+      # Return information about a single table for the given storage
+      # account +key+. If you are looking for the entities within the
+      # table, use the table_data method instead.
+      #
+      def table_info(table, key = nil)
```
Comment (bzwei) [inline lib/azure/armrest/model/storage_account.rb]: And this method to be `get_table`

File: lib/azure/armrest/model/storage_account.rb [hunk 3]
```diff
@@ -24,6 +28,52 @@ def initialize(json)
...
+        key ||= properties.key1
+        response = table_response(key, nil, "Tables('#{table}')")
+        Table.new(response.body)
+      end
+
+      # Returns a list of TableData objects for the given table +name+ using
+      # account +key+. The exact nature of the TableData object depends on the
+      # type of table that it is.
+      #
+      def table_data(name, key = nil, options = {})
```
Comment (bzwei) [inline lib/azure/armrest/model/storage_account.rb]: `list_table_data`?

**Comment by djberg96** [comment]
@bzwei I didn't follow that convention here because it's not a Service class, and it felt clumsy. I didn't follow that convention for any of the blob methods, and those methods are already being used by MIQ.

I could create aliases, though.

**Comment by bzwei** [comment]
@djberg96 right, I did not realize this is not the service class. Alias methods are not necessary then.

**Pull request merged**